orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/V2SchemaExecutionGenerator.java

```diff
@@ -110,8 +111,25 @@ private void addTriggers(
       pipeline.put(
           "triggers",
           TemplateMerge.mergeDistinct(
-              (List<HashMap<String, Object>>) template.getPipeline().get("triggers"),
+              markInherited(getTemplateCollection(template, "triggers")),
               configuration.getTriggers()));
     }
   }
+
+  private static List<HashMap<String, Object>> markInherited(
```

Comment:
```suggestion
  private static List<Map<String, Object>> markInherited(
```

Reply:
I think we typically want to have return types be interfaces unless there's a good reason for them to be implementations

Reply:
Separately, I feel like the name `markInherited` incorrectly implies that we are mutating the input. I don't have a good alternative name though...

Reply:
`getInheritanceDesignatedTemplateCollections`? Probably too verbose, ignore me :see_no_evil:

Reply:
👍 I agree that markInherited is a bit confusing. I am renaming and refactoring to make it clearer that it mutates with an `inherited` key

Reply:
Nice LGTM :100: